Spawning anonymous functions that have no input arguments  (example from book: Getting Started with V Programming) fails

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.10 52c7130, press to see full `v doctor` output</summary>

|V full version      |V 0.4.10 22d2d1a.52c7130
|:-------------------|:-------------------
|OS                  |linux, "Manjaro Linux"
|Processor           |16 cpus, 64bit, little endian, AMD Ryzen 9 7940HS w/ Radeon 780M Graphics
|Memory              |13.68GB/30.57GB
|                    |
|V executable        |/home/pepe/Dev/v/v
|V last modified time|2025-05-25 10:55:16
|                    |
|V home dir          |OK, value: /home/pepe/Dev/v
|VMODULES            |OK, value: /home/pepe/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/pepe/_mysrc/test/vlang
|                    |
|Git version         |git version 2.49.0
|V git status        |weekly.2025.21-26-g52c7130a
|.git/config present |true
|                    |
|cc version          |cc (GCC) 15.1.1 20250425
|gcc version         |gcc (GCC) 15.1.1 20250425
|clang version       |clang version 19.1.7
|tcc version         |tcc version 0.9.28rc 2024-02-05 HEAD@105d70f7 (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 a0799a5b
|emcc version        |N/A
|glibc version       |ldd (GNU libc) 2.41

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  concurrency06.v && concurrency06`
```v
module main

fn main() {
	t := go fn () string {
		return 'hi'
	}()
	x := t.wait()
	println(typeof(x).name) // string
	println(x) // hi
}

```

**What did you see?**
```
Compilation with unknown C compiler
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/concurrency06.01JW4FBQ8TPYRJTYPD78066AW9.tmp.c:1290: warning: implicit declaration of function '_v_free'
cc: tcc: error: undefined symbol '_v_free'
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

**What did you expect to see?**

'hi'

Also on Manjaro here, and I get a 2 error messages...
```
/tmp/v_1000/x.01JW4H6S1N8GXSZPJT9QD4J909.tmp.c:1293: warning: implicit declaration of function '_v_free'
/home/jalon/git/v/thirdparty/tcc/lib/libgc.a: error: 'GC_noop1_ptr' defined twice
tcc: error: undefined symbol '_v_free'
```

--- a/vlib/v/ast/table.v
+++ b/vlib/v/ast/table.v
@@ -32,6 +32,7 @@ pub mut:
 	map_update       bool            // {...foo}
 	interpolation    bool            // '${foo} ${bar}'
 	option_or_result bool            // has panic call
+	waiter           bool            // has thread waiter
 	print_types      map[int]bool    // print() idx types
 	used_fns         map[string]bool // filled in by markused
 	used_consts      map[string]bool // filled in by markused

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -2726,6 +2726,9 @@ fn (mut c Checker) spawn_expr(mut node ast.SpawnExpr) ast.Type {
 		c.error('option handling cannot be done in `spawn` call. Do it when calling `.wait()`',
 			node.call_expr.or_block.pos)
 	}
+	if node.is_expr {
+		c.table.used_features.waiter = true
+	}
 	// Make sure there are no mutable arguments
 	for arg in node.call_expr.args {
 		if arg.is_mut && !arg.typ.is_ptr() {
@@ -2760,6 +2763,9 @@ fn (mut c Checker) go_expr(mut node ast.GoExpr) ast.Type {
 	if node.call_expr.or_block.kind != .absent {
 		c.error('option handling cannot be done in `go` call. Do it when calling `.wait()`',
 			node.call_expr.or_block.pos)
+	}
+	if node.is_expr {
+		c.table.used_features.waiter = true
 	}
 	// Make sure there are no mutable arguments
 	for arg in node.call_expr.args {

--- a/vlib/v/markused/markused.v
+++ b/vlib/v/markused/markused.v
@@ -168,6 +168,9 @@ pub fn mark_used(mut table ast.Table, mut pref_ pref.Preferences, ast_files []&a
 				builderptr_idx + '.write_rune',
 			]
 		}
+		if table.used_features.waiter {
+			core_fns << 'free'
+		}
 		if !table.used_features.arr_init {
 			table.used_features.arr_init = table.used_features.print_types.keys().any(table.type_to_str(it).contains('[]'))
 		}

--- a/vlib/v/tests/skip_unused/thread_waiter.run.out
+++ b/vlib/v/tests/skip_unused/thread_waiter.run.out
@@ -1,0 +1,2 @@
+string
+hi

--- a/vlib/v/tests/skip_unused/thread_waiter.skip_unused.run.out
+++ b/vlib/v/tests/skip_unused/thread_waiter.skip_unused.run.out
@@ -1,0 +1,2 @@
+string
+hi

--- a/vlib/v/tests/skip_unused/thread_waiter.vv
+++ b/vlib/v/tests/skip_unused/thread_waiter.vv
@@ -1,0 +1,10 @@
+module main
+
+fn main() {
+	t := go fn () string {
+		return 'hi'
+	}()
+	x := t.wait()
+	println(typeof(x).name)
+	println(x)
+}

